SeleniumAutomationGenerator/ComponentFactorySteps/FileCreatorFactoryStep.cs

```diff
@@ -0,0 +1,82 @@
+
+        public FileCreatorFactoryStep(IFileCreatorContainer container, IClassAppenderContainer appendersContainer, IAddinContainer addinsContainer) : base(appendersContainer)
+        {
+            _container = container;
+            _addinsContainer = addinsContainer;
+        }
+
+        public override bool ShouldInvokeStep(AutoElementData rootElement, IComponentFileCreator parent)
+        {
+            string type = SelectorUtils.GetKeyWordFromSelector(rootElement.Selector);
+            return _container.GetFileCreator(type) != null;
+        }
+
+        public override IEnumerable<ComponentGeneratorOutput> InvokeStep(AutoElementData rootElement, IComponentFileCreator parent, IComponentFactoryCreatorStep next = null)
+        {
+            string selector = rootElement.Selector;
+            AutoElementData[] filteredChildren = rootElement.InnerChildrens
+                .Where(elm => !IsAppenderElement(elm))
+                .ToArray();
+            string keyWord = SelectorUtils.GetKeyWordFromSelector(selector);
+            return GenerateAppendersOutputs(keyWord, rootElement.InnerChildrens, next)
+                            .Union(GenerateFileCreatorOutputs(selector, filteredChildren, keyWord, next), new ComponentOutputComparer());
+        }
+
+        private IEnumerable<ComponentGeneratorOutput> GenerateAppendersOutputs(string keyword, IEnumerable<AutoElementData> elements, IComponentFactoryCreatorStep next)
+        {
+            IComponentFileCreator parent = _container.GetFileCreator(keyword);
+            IEnumerable<AutoElementData> appendersElements = elements
+                            .Where(IsAppenderElement);
+            return GenerateClassesForElements(appendersElements, parent, next);
+        }
+
+        private IEnumerable<ComponentGeneratorOutput> GenerateFileCreatorOutputs(string selector,
+            AutoElementData[] children, string keyWord, IComponentFactoryCreatorStep next)
+        {
+            ElementSelectorData[] elements = children
+                            .Select(ConversionsUtils.ConvertToElementSelectorData)
+                            .Select(TransformFileCreatorToAddinsLike)
+                            .ToArray();
+
+            IComponentFileCreator current = _container.GetFileCreator(keyWord);
+            IEnumerable<ComponentGeneratorOutput> outputs = GenerateClassesForElements(children, current, next);
+
+            ComponentGeneratorOutput parentOutput = current.GenerateComponentClass(selector, elements);
+            outputs = outputs.Union(new[] { parentOutput }, new ComponentOutputComparer());
+
+            return outputs;
+        }
+
+        private ElementSelectorData TransformFileCreatorToAddinsLike(ElementSelectorData child)
+        {
+            if (!IsAddinElement(child))
+            {
+                IComponentFileCreator fileCreator = _container.GetFileCreator(child.Type);
+                if (fileCreator != null)
```

Comment:
`fileCreator` is never null. if is doesn't find a creator it returns the default. (maybe create a `NullFileCreator`?)